### PR TITLE
OSDOCS-5340 Adding categories and projects to Nutanix install doc

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.13

Issue:
https://issues.redhat.com/browse/OSDOCS-5340

Link to docs preview:
[Installing a cluster on Nutanix (table 4)](https://57130--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-nutanix-installer-provisioned.html#installation-configuration-parameters_installing-nutanix-installer-provisioned)
[Installing a cluster on Nutanix (sample yaml file)](https://57130--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-nutanix-installer-provisioned.html#installation-nutanix-config-yaml_installing-nutanix-installer-provisioned)
[Restricted cluster (table 4)](https://57130--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-restricted-networks-nutanix-installer-provisioned.html#installation-configuration-parameters_installing-restricted-networks-nutanix-installer-provisioned)
[Restricted cluster (sample yaml file)](https://57130--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-restricted-networks-nutanix-installer-provisioned.html#installation-nutanix-config-yaml_installing-restricted-networks-nutanix-installer-provisioned)

QE review:
- [X] QE has approved this change.

### CODE REVIEW

**Comment by bscott-rh** [comment]
@jianlinliu Hello, please take a look at the documentation update for https://github.com/openshift/release/pull/37029. Thank you

**Comment by jianlinliu** [comment]
@sgaoshang pls help review the PR.

**Comment by sgaoshang** [comment]
@bscott-rh most lgtm, only that 3 new fields are added in [PR](https://github.com/openshift/installer/pull/6841), so should we also address `bootType` here? Also pls refer to https://redhat-internal.slack.com/archives/C0211848DBN/p1678126369498519 for bootType field.
```
The new fields include:
bootType: optionally indicates the boot type for the worker Machine VM.
project: optionally identifies a Prism project for the Machine's VM.
categories: optionally adds one or more prism categories for the Machine's VM.
```

**Comment by bscott-rh** [comment]
> @bscott-rh most lgtm, only that 3 new fields are added in [PR](https://github.com/openshift/installer/pull/6841), so should we also address `bootType` here? Also pls refer to https://redhat-internal.slack.com/archives/C0211848DBN/p1678126369498519 for bootType field.
> 
> ```
> The new fields include:
> bootType: optionally indicates the boot type for the worker Machine VM.
> project: optionally identifies a Prism project for the Machine's VM.
> categories: optionally adds one or more prism categories for the Machine's VM.
> ```

Thanks for catching this, I noted it in Slack but didn't note it in real life :) updated the PR, PTAL. Can you check the Nutanix documentation link that I referenced for bootType and let me know if you know of a better link I can share to explain the boot modes?

**Comment by sgaoshang** [comment]
> > @bscott-rh most lgtm, only that 3 new fields are added in [PR](https://github.com/openshift/installer/pull/6841), so should we also address `bootType` here? Also pls refer to https://redhat-internal.slack.com/archives/C0211848DBN/p1678126369498519 for bootType field.
> > ```
> > The new fields include:
> > bootType: optionally indicates the boot type for the worker Machine VM.
> > project: optionally identifies a Prism project for the Machine's VM.
> > categories: optionally adds one or more prism categories for the Machine's VM.
> > ```
> 
> Thanks for catching this, I noted it in Slack but didn't note it in real life :) updated the PR, PTAL. Can you check the Nutanix documentation link that I referenced for bootType and let me know if you know of a better link I can share to explain the boot modes?

I did not find a better link maybe `5 Select one of the following firmware to boot the VM` in https://portal.nutanix.com/page/documents/details?targetId=AHV-Admin-Guide:wc-vm-create-acropolis-wc-t.html?

Others lgtm, thanks.

**Comment by yannickstruyf3** [comment]
LGTM

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -1831,6 +1831,62 @@ Additional Nutanix configuration parameters are described in the following table
...
+|`compute.platform.nutanix.categories.key`
+|The name of a prism category key to apply to compute VMs. This parameter must be accompanied by the `value` parameter, and both `key` and `value` parameters must exist in Prism Central. For more information on categories, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_6:ssp-ssp-categories-manage-pc-c.html[Category management].
+|String
+
+|`compute.platform.nutanix.categories.value`
+|The value of a prism category key-value pair to apply to compute VMs. This parameter must be accompanied by the `key` parameter, and both `key` and `value` parameters must exist in Prism Central.
+|String
+
+|`compute.platform.nutanix.project.type`
+|The type of identifier you use to select a project for compute VMs.  Projects define logical groups of user roles for managing permissions, networks and other parameters. For more information on projects, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_9:ssp-app-mgmt-project-env-c.html[Projects Overview].
```
Comment (mburke5678) [inline modules/installation-configuration-parameters.adoc:1843]: s/networks/networks, (Oxford comma)

File: modules/installation-configuration-parameters.adoc [hunk 2]
```diff
@@ -1831,6 +1831,62 @@ Additional Nutanix configuration parameters are described in the following table
...
+|`name` or `uuid`
+
+|`compute.platform.nutanix.project.name` or `compute.platform.nutanix.project.uuid`
+|The name or UUID of a project with which compute VMs are associated. This parameter must be accompanied by the `type` parameter.
+|String
+
+|`compute.platform.nutanix.bootType`
+|The boot type that compute machines use. You must use `Legacy` boot type in {product-title} {product-version}. For more information on boot types, see link:https://portal.nutanix.com/page/documents/kbs/details?targetId=kA07V000000H3K9SAK[Understanding UEFI, Secure Boot, and TPM in the Virtualized Environment].
```
Comment (mburke5678) [inline modules/installation-configuration-parameters.adoc:1851]: s/that compute machines/that the compute machines 
s/`Legacy`/the `Legacy`

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -1831,6 +1831,62 @@ Additional Nutanix configuration parameters are described in the following table
...
+|`controlPlane.platform.nutanix.categories.key`
+|The name of a prism category key to apply to control plane VMs. This parameter must be accompanied by the `value` parameter, and both `key` and `value` parameters must exist in Prism Central. For more information on categories, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_6:ssp-ssp-categories-manage-pc-c.html[Category management].
+|String
+
+|`controlPlane.platform.nutanix.categories.value`
+|The value of a prism category key-value pair to apply to control plane VMs. This parameter must be accompanied by the `key` parameter, and both `key` and `value` parameters must exist in Prism Central.
+|String
+
+|`controlPlane.platform.nutanix.project.type`
+|The type of identifier you use to select a project for control plane VMs.  Projects define logical groups of user roles for managing permissions, networks and other parameters. For more information on projects, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_9:ssp-app-mgmt-project-env-c.html[Projects Overview].
```
Comment (mburke5678) [inline modules/installation-configuration-parameters.adoc:1863]: s/networks/networks, (Oxford comma)

File: modules/installation-configuration-parameters.adoc [hunk 4]
```diff
@@ -1831,6 +1831,62 @@ Additional Nutanix configuration parameters are described in the following table
...
+|`platform.nutanix.defaultMachinePlatform.categories.key`
+|The name of a prism category key to apply to all VMs. This parameter must be accompanied by the `value` parameter, and both `key` and `value` parameters must exist in Prism Central. For more information on categories, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_6:ssp-ssp-categories-manage-pc-c.html[Category management].
+|String
+
+|`platform.nutanix.defaultMachinePlatform.categories.value`
+|The value of a prism category key-value pair to apply to all VMs. This parameter must be accompanied by the `key` parameter, and both `key` and `value` parameters must exist in Prism Central.
+|String
+
+|`platform.nutanix.defaultMachinePlatform.project.type`
+|The type of identifier you use to select a project for all VMs. Projects define logical groups of user roles for managing permissions, networks and other parameters. For more information on projects, see link:https://portal.nutanix.com/page/documents/details?targetId=Prism-Central-Guide-vpc_2022_9:ssp-app-mgmt-project-env-c.html[Projects Overview].
```
Comment (mburke5678) [inline modules/installation-configuration-parameters.adoc:1879]: s/networks/networks, (Oxford comma)

File: modules/installation-configuration-parameters.adoc [hunk 5]
```diff
@@ -1831,6 +1831,62 @@ Additional Nutanix configuration parameters are described in the following table
...
+|`name` or `uuid`.
+
+|`platform.nutanix.defaultMachinePlatform.project.name` or `platform.nutanix.defaultMachinePlatform.project.uuid`
+|The name or UUID of a project with which all VMs are associated. This parameter must be accompanied by the `type` parameter.
+|String
+
+|`platform.nutanix.defaultMachinePlatform.bootType`
+|The boot type for all machines. You must use `Legacy` boot type in {product-title} {product-version}. If you use `SecureBoot`, `UEFI` will also be enabled. For more information on boot types, see link:https://portal.nutanix.com/page/documents/kbs/details?targetId=kA07V000000H3K9SAK[Understanding UEFI, Secure Boot, and TPM in the Virtualized Environment].
```
Comment (mburke5678) [inline modules/installation-configuration-parameters.adoc:1887]: s/Legacy/the Legacy

> If you use `SecureBoot`,
The preceding sentence says I must use `Legacy`. Which is correct?

File: modules/installation-nutanix-config-yaml.adoc [hunk 6]
```diff
@@ -98,20 +112,22 @@ endif::openshift-origin[]
 If you disable simultaneous multithreading, ensure that your capacity planning accounts for the dramatically decreased machine performance.
 ====
 <4> Optional: Provide additional configuration for the machine pool parameters for the compute and control plane machines.
-<5> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
+<5> Optional: provide one or more pairs of a prism category key and a prism category value. These category key-value pairs must exist in Prism Central. You can provide separate categories to compute machines, control plane machines, or all machines.
```
Comment (mburke5678) [inline modules/installation-nutanix-config-yaml.adoc:115]: Capitalize the first word after Optional:

File: modules/installation-nutanix-config-yaml.adoc [hunk 7]
```diff
@@ -98,20 +112,22 @@ endif::openshift-origin[]
...
+<6> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
 ifndef::openshift-origin[]
-<6> Optional: By default, the installation program downloads and installs the {op-system-first} image. If Prism Central does not have internet access, you can override the default behavior by hosting the {op-system} image on any HTTP server and pointing the installation program to the image.
-<7> Whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the cryptography modules that are provided with {op-system} instead.
+<7> Optional: specify a project with which VMs are associated. Specify either `name` or `uuid` for the project type, and then provide the corresponding UUID or project name. You can associate projects to compute machines, control plane machines, or all machines.
```
Comment (mburke5678) [inline modules/installation-nutanix-config-yaml.adoc:118]: Capitalize the first word after Optional:

File: modules/installation-nutanix-config-yaml.adoc [hunk 8]
```diff
@@ -223,39 +253,41 @@ endif::openshift-origin[]
...
 ====
 <4> Optional: Provide additional configuration for the machine pool parameters for the compute and control plane machines.
-<5> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
-<6> Optional: By default, the installation program downloads and installs the {op-system-first} image. If Prism Central does not have internet access, you can override the default behavior by hosting the {op-system} image on any HTTP server or Nutanix Objects and pointing the installation program to the image.
-<7> For `<local_registry>`, specify the registry domain name, and optionally the port, that your mirror registry uses to serve content. For example `registry.example.com` or `registry.example.com:5000`. For `<credentials>`,
+<5> Optional: provide one or more pairs of a prism category key and a prism category value. These category key-value pairs must exist in Prism Central. You can provide separate categories to compute machines, control plane machines, or all machines.
```
Comment (mburke5678) [inline modules/installation-nutanix-config-yaml.adoc:256]: Capitalize the first word after Optional:

File: modules/installation-nutanix-config-yaml.adoc [hunk 9]
```diff
@@ -223,39 +253,41 @@ endif::openshift-origin[]
...
+<6> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
+<7> Optional: specify a project with which VMs are associated. Specify either `name` or `uuid` for the project type, and then provide the corresponding UUID or project name. You can associate projects to compute machines, control plane machines, or all machines.
```
Comment (mburke5678) [inline modules/installation-nutanix-config-yaml.adoc:258]: Capitalize the first word after Optional:

**Comment by mburke5678** [comment]
@bscott-rh A few nits. Otherwise LGTM

File: modules/installation-configuration-parameters.adoc (hunk 5 above)
Comment (bscott-rh) [reply modules/installation-configuration-parameters.adoc:1887]: Good catch, I meant to remove that sentence. Thanks

**Pull request merged**